apm-lambda-extension/apmproxy/receiver.go

```diff
@@ -121,17 +121,32 @@ func (c *Client) handleIntakeV2Events() func(w http.ResponseWriter, r *http.Requ
 			return
 		}
 
-		if len(rawBytes) > 0 {
-			agentData := AgentData{
-				Data:            rawBytes,
-				ContentEncoding: r.Header.Get("Content-Encoding"),
-			}
+		flushed := r.URL.Query().Get("flushed") == "true"
 
-			c.EnqueueAPMData(agentData)
+		agentData := AgentData{
+			Data:            rawBytes,
+			ContentEncoding: r.Header.Get("Content-Encoding"),
 		}
 
-		if len(r.URL.Query()["flushed"]) > 0 && r.URL.Query()["flushed"][0] == "true" {
-			c.AgentDoneSignal <- struct{}{}
+		enqueued := c.EnqueueAPMData(agentData)
+
+		if enqueued && flushed {
```

Comment:
```suggestion
		if agentFlushed {
```

Closing the channel shouldn't depend on any enqueuing.